### PR TITLE
Access & Review request docs

### LINKED ISSUE #7135: Support for access requests, reviews specifications in roles for variable substitution
### What
Support  usage of `regexp` functions and external traits in requests and reviews in roles.

```yaml
    request:
      roles:
      - '{{regexp.replace(user.spec.traits[groups-request], ^product(.*)$, "$1-admin")}}'
      - '{{external.additionalroles}}'

    review_requests:
      roles:
     - '{{regexp.replace(user.spec.traits[groups-request], ^product(.*)$, "$1-admin")}}'
```
### How
Extend current role mapping to include these parts of the role to allow usage of functions and external traits.

### Why
Allows for doing dynamic mapping in roles with the traits coming from the SSO>

### Workaround
Manually specifying in roles

### PR DESCRIPTION
**Pull request by tcsc**
Adds documentation on how to use `claims_to_roles` in Access and Review Requests.

This is very much a first pass as I have worked out most of it by reading the code & lots of `printf()`s. Edits & suggestions most welcome.

See-Also: #7135
See-Also: #7538

### CODE REVIEW

**Comment by russjones** [comment]
@stevenGravy Can you review? This the the Access Requests variable interpolation issue. I was already implemented just not documented.

**Review by stevenGravy — APPROVED** [review]
LGTM

**Comment by tcsc** [comment]
Obsoleted by #7791

**Comment by tcsc** [comment]
@klizhentas - can I please get a quick UX review on this please?

File: docs/pages/access-controls/guides/role-templates.mdx [hunk 1]
```diff
@@ -325,3 +325,47 @@ Function | Description
...
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: product-admin
+spec:
+  allow:
+    request:
+      # `roles` is a static list of roles a user with the `peodyct-admin` role may
```
Comment (klizhentas) [inline docs/pages/access-controls/guides/role-templates.mdx:345]: `peodyct-admin` -> `product-admin`

File: docs/pages/access-controls/guides/role-templates.mdx [hunk 2]
```diff
@@ -325,3 +325,47 @@ Function | Description
 --- | ---
 `email.local(variable)` | Extracts the local part of an email field, like `Alice <alice@example.com>` or `bob@example.com`.
 `regexp.replace(variable, expression, replacement)` | Finds all matches of `expression` and replaces them with `replacement`. This supports expansion, e.g. `regexp.replace(external.email, "^(.*)@example.com$", "$1")`. Values which do not match the expression will be filtered out.
+
+## Pattern matching in Access & Review Requests
```
Comment (klizhentas) [inline docs/pages/access-controls/guides/role-templates.mdx:329]: Let's shorten this to `Templating in Access Requests`

File: docs/pages/access-controls/guides/role-templates.mdx (hunk 1 above)
Comment (tcsc) [reply docs/pages/access-controls/guides/role-templates.mdx]: Not sure what language that was. Fixed.

File: docs/pages/access-controls/guides/role-templates.mdx (hunk 2 above)
Comment (tcsc) [reply docs/pages/access-controls/guides/role-templates.mdx]: Done

**Review by klizhentas — APPROVED** [review]
